fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/filter/TenantAwareBasicAuthenticationFilter.java

```diff
@@ -59,33 +58,34 @@
  *
  * If multi-tenant and basic auth credentials are invalid, a http error response is returned.
  */
-@Service
+
 @Profile("basicauth")
 public class TenantAwareBasicAuthenticationFilter extends BasicAuthenticationFilter {
 
     private static boolean firstRequestProcessed = false;
     private static final Logger LOG = LoggerFactory.getLogger(TenantAwareBasicAuthenticationFilter.class);
 
-    private final BasicAuthTenantDetailsService basicAuthTenantDetailsService;
-    private final ToApiJsonSerializer<PlatformRequestLog> toApiJsonSerializer;
-    private final ConfigurationDomainService configurationDomainService;
-    private final CacheWritePlatformService cacheWritePlatformService;
-    private final NotificationReadPlatformService notificationReadPlatformService;
+    @Autowired
```

Comment:
A compromise here could be to use Lombok with "RequiredArgsConstructor"... then we can keep the same dependency injection strategy without being annoying (aka having to write all the constructor boilerplate code). Personally the "@Autowired" are OK for me... not sure if others have an opinion here.

If we introduce Lombok (I really would like) then I prefer the combination of "@RequiredArgsConstructor" and final class attributes.

My 2 cents.

Reply:
I think this is a case of personal preferences... personally I like declaring the variables as @Autowired more than the "huge constructor with lots of parameters" approach, as it's simply less code: you just need to declare the variables, not repeat them three times (one declaration and twice in constructor). But you're right that it differs from how most of the classes are currently written...